src/uu/cp/src/platform/other_unix.rs

```diff
@@ -47,6 +48,16 @@ pub(crate) fn copy_on_write(
             .mode(mode)
             .open(dest)?;
 
+        let dest_filetype = dst_file.metadata()?.file_type();
+        let dest_is_stream = dest_filetype.is_fifo()
```

Comment:
seems that the code is duplicated, could you please move that into a function? thanks